pipeline/post-results.groovy

```diff
@@ -0,0 +1,61 @@
+/*
+    Pipeline script for executing Tier 0 test suites for RH Ceph Storage in IBM cloud.
+*/
+
+def nodeName = "centos-7"
+def credsRpProc = [:]
+def sharedLib
+def rpPreprocDir
+def reportBucket = "qe-ci-reports"
+def remoteName= "ibm-cos"
+
+node(nodeName) {
+
+    timeout(unit: "MINUTES", time: 30) {
+        stage('Install prereq') {
+            if (env.WORKSPACE) {
+                sh script: "sudo rm -rf *"
+            }
+            checkout([
+                $class: 'GitSCM',
+                branches: [[name: 'origin/master']],
+                doGenerateSubmoduleConfigurations: false,
+                extensions: [
+                    [
+                        $class: 'CloneOption',
+                        shallow: true,
+                        noTags: true,
+                        reference: '',
+                        depth: 1
+                    ],
+                    [$class: 'CleanBeforeCheckout'],
+                ],
+                submoduleCfg: [],
+                userRemoteConfigs: [[
+                    url: 'https://github.com/red-hat-storage/cephci.git'
+                ]]
+            ])
+
+            // prepare the node
+            sharedLib = load("${env.WORKSPACE}/pipeline/vars/lib.groovy")
+            sharedLib.prepareNode()
+        }
+    }
+
+    stage('Configure RP_PreProc'){
+        (rpPreprocDir, credsRpProc) = sharedLib.configureRpPreProc()
+    }
+
+    stage('Upload xml to report-portal'){
+        def configFile = sh(script: "rclone config file", returnStdout: true )
+        def listFiles = sh(
+            script: "rclone lsf ${remoteName}:${reportBucket} -R --files-only",
+            returnStdout: true
+        )
+        def fileNames = listFiles.split("\\n")
```

Comment:
IMO, we can avoid multiple calls to the object store by performing
```
1) Pull all the objects to a temporary dir by 'rclone sync qe-ci-reports <temporary_dir>'
2) List all files in <temporary_dir>
3) for each file in the folder
- upload the information to ReportPortal
- remove the file from COS
```

Reply:
rclone sync would download multiple test run directories with xml files in it (i.e, /temp/<test-run-dir>/files).
So currently, getting all files list at once gives easy way to process, because iterating through all xml files will never be removed to protect launch name.

Reply:
Okay.